### PR TITLE
Add `loop`, `break` and `continue`

### PR DESCRIPTION
**Pull request by kubouch**
Add `loop`, `break` and `continue` statements.

Also fixes `for` to accept declarations without dollar sign, e.g., `for i in 0..10 { ... }`.

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -1195,7 +1206,7 @@ impl Parser {
         let span_start = self.position();
         self.keyword(b"for");
 
-        let variable = self.variable();
+        let variable = self.variable_decl();
```
Comment (sholderbach) [inline src/parser.rs:1209]: Good call. I personally would favor us consolidating towards the binding without `$` but that can be discussed later.

File: src/parser.rs [hunk 2]
```diff
@@ -1213,6 +1224,15 @@ impl Parser {
         )
     }
 
+    pub fn loop_statement(&mut self) -> NodeId {
+        let span_start = self.position();
+        self.keyword(b"loop");
+        let block = self.block(BlockContext::Curlies);
+        let span_end = self.get_span_end(block);
```
Comment (sholderbach) [inline src/parser.rs:1231]: Looking good but a quesiton we should discuss with @sophiajt is how we want to consistently track the span ends. 
We have: 
- peeking the inner parse like here via `self.get_span_end()`
- manual computation (based on known lengths or byte offsets)
- just querying `self.position()` after the parse

**Review by sholderbach — APPROVED** [review]
Looking good! Thanks for all the good test cases!

File: src/parser.rs (hunk 2 above)
Comment (kubouch) [reply src/parser.rs:1231]: Yeah, I found this slightly awkward as well. Maybe `self.get_span(span_start)` could be added, which would internally use `self.position()` and create a span using the given span start?

**Comment by sophiajt** [comment]
@sholderbach @kubouch - y'all should run the tests ;)

Seems we have a few broken ones related to the new keywords. Easy to fix but maybe we should have some simple CI on this repo.

**Comment by kubouch** [comment]
I ran the tests but forgot to update the snaps, dang it.

**Pull request merged**